wikimedia/wikipedia-ios
# Localize image insert wikitext

**Phabricator:** https://phabricator.wikimedia.org/T364408

### Notes
This PR adds in the magic words logic (related [spike](https://phabricator.wikimedia.org/T355326#9536130)) into our image insert flow. This change is applied in both image recommendations and the editor. It also removes some incorrect parameters.

Similar to what we were already doing with namespace names, I am bundling these magic words into the project. To do this I updated the Update Languages script to fetch from the `siteinfo` API and added the json files into the `wikipedia-magicwords` directory.

### Test Steps
The easiest way to test this is through the editor, without publishing.

Stripping default size:
1. Go to editor on EN Wikipedia.
2. Tap insert image button
3. Tap image, leave advanced settings as default, and tap through to insert image.
4. Confirm wikitext inserted into editor does not contain size.

Stripping "basic"
1. Go to editor on EN Wikipedia.
2. Tap insert image button
3. Tap image, go to advanced settings and select "basic", and tap through to insert image.
4. Confirm wikitext inserted into editor does not contain word "basic".

Localizing wikitext to use magic words
1. Go to editor on DE Wikipedia.
2. Tap insert image button
3. Tap image, leave advanced settings as default, add alt text and caption, and tap through to insert image.
4. Confirm wikitext inserted into editor uses localized template parameters and namespace:

`[[Datei:Biblioteca Nacional - Busto de German Arciniegas.jpg | mini | rechts | alternativtext=Alt Text | Caption]]`
